lib/rules/selector-no-deprecated/index.mjs

```diff
@@ -0,0 +1,167 @@
+import selectorParser from 'postcss-selector-parser';
+const { isPseudo, isPseudoElement, isTag } = selectorParser;
+
+import {
+	deprecatedHtmlTypeSelectors,
+	deprecatedPseudoClasses,
+	deprecatedPseudoElements,
+	deprecatedSvgTypeSelectors,
+} from '../../reference/selectors.mjs';
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import getRuleSelector from '../../utils/getRuleSelector.mjs';
+import isCustomSelector from '../../utils/isCustomSelector.mjs';
+import isStandardSyntaxRule from '../../utils/isStandardSyntaxRule.mjs';
+import isStandardSyntaxSelector from '../../utils/isStandardSyntaxSelector.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import parseSelector from '../../utils/parseSelector.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import uniteSets from '../../utils/uniteSets.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'selector-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	expected: (unfixed, fixed) => `Expected "${unfixed}" to be "${fixed}"`,
+	rejected: (selector) => `Unexpected deprecated selector "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/selector-no-deprecated',
+	fixable: true,
+};
+
+/** @type {Partial<Record<string, string | null>>} */
+const DEPRECATED_PSEUDO_CLASSES_REMAP = {
+	'focus-ring': 'focus-visible',
+	matches: 'is',
+};
+
```

Comment:
suggestion
minor

If we have `matches` it would make sense to add these

```
const Foo = uniteSets(deprecatedPseudoClasses, [
  '-moz-any',
  '-webkit-any',
];
```

i.e. not directly in `selectors.mjs`